### PR TITLE
Initial commit with 24 Actions added.

### PR DESCRIPTION
**Pull request by cs-adarsh-gupta**
#### UTCs: 
 
- [x] Connector installation verified. 
- [x] Connector logo verified. 
- [x] Actions and Playbook list verified. 
- [x] Playbook tags verified. 
- [x] All Playbook are in info mode verified. 
- [x] All Playbook are in inactive mode verified.

### CODE REVIEW

File: netbox/info.json [hunk 1]
```diff
@@ -0,0 +1,2668 @@
...
+          "value": {
+            "address": "string",
+            "vrf": 0,
+            "tenant": 0,
+            "status": "active",
+            "role": "loopback",
+            "assigned_object_type": "string",
+            "assigned_object_id": 9223372036854776000,
+            "nat_inside": 0,
+            "dns_name": "*.G7tafc4lbh4wzM6rk8u4RVrwSZQjKHUkopm6odkzY7ao3A5D6ZBtPlhdsXOVwGSq3H9vuWbb.D5EWQI0yFjpnkXiBKJpSObjpNovFX8-.F4VlX_dXq0QVuGYb6vCfdm.fn8ByqgCQBiD44X2qBoT6bQ_0ziYqaTedzFD4UK9lxhrh9ay_m0cHf9u5FYiUjRkEMIEcSEPRvlp.KR8hmwgztO7ELv-b.W4uo1ZOWY99FKQfaztOyjuYb2O6j4OE",
```
Comment (cs-sameer-pathan) [inline netbox/info.json]: Please remove this example or keep keys only with empty values.
Provide actual example in sample playbook

File: netbox/info.json [hunk 2]
```diff
@@ -0,0 +1,2668 @@
...
+          "name": "limit",
+          "visible": true,
+          "required": false,
+          "editable": true,
+          "type": "integer",
+          "tooltip": "Specify the Number of results to return per page.",
+          "description": "Specify the Number of results to return per page."
+        },
+        {
+          "title": "Other Fields",
```
Comment (cs-sameer-pathan) [inline netbox/info.json:211]: Please add important parameters like id , status, filter etc. if possible in other actions as well

File: netbox/operations.py [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+import requests
+from connectors.core.connector import get_logger, ConnectorError
+
+logger = get_logger('netbox')
+
+
+class Netbox:
+    def __init__(self, config):
+        self.server_url = config.get('server_url')
+        if not self.server_url.startswith('https://'):
```
Comment (cs-sameer-pathan) [inline netbox/operations.py]: Please check for http or https

File: netbox/operations.py [hunk 4]
```diff
@@ -0,0 +1,428 @@
...
+            self.server_url = 'https://' + self.server_url
+        self.server_url.strip('/')
```
Comment (cs-sameer-pathan) [inline netbox/operations.py]: need to assign this value to self.server_url

File: netbox/operations.py [hunk 5]
```diff
@@ -0,0 +1,428 @@
...
+        self.api_key = config.get('api_key')
+        self.verify_ssl = config.get('verify_ssl')
+
+    def make_request(self, endpoint, method='get', data=None, params=None, files=None):
+        try:
+            url = self.server_url + endpoint
+            logger.info('Executing url {}'.format(url))
+            headers = {'Authorization': f"Token {self.api_key}", 'Content-Type': 'application/json'}
+
+            # # CURL UTILS CODE
```
Comment (cs-sameer-pathan) [inline netbox/operations.py]: why commented this section ?

File: netbox/operations.py [hunk 6]
```diff
@@ -0,0 +1,428 @@
...
+            #     from connectors.debug_utils.curl_script import make_curl
+            #     make_curl(method, endpoint, headers=headers, params=params, data=data, verify_ssl=self.verify_ssl)
+            # except Exception as err:
+            #     logger.error(f"Error in curl utils: {str(err)}")
+
+            response = requests.request(method, url, params=params, files=files, data=data, headers=headers,
+                                        verify=self.verify_ssl)
+            if response.ok:
+                logger.info('successfully get response for url {}'.format(url))
+                if method == 'delete':
```
Comment (cs-sameer-pathan) [inline netbox/operations.py]: Please check case of the string 'delete' or check with method.lower() == 'delete'

File: netbox/operations.py [hunk 7]
```diff
@@ -0,0 +1,428 @@
...
+    except Exception as err:
+        logger.error(str(err))
+        raise ConnectorError(str(err))
+
+
+def _check_health(config):
+    try:
+        tg = Netbox(config)
+        endpoint = '/api/dcim/sites/'
+        response = tg.make_request(endpoint=endpoint)
```
Comment (cs-sameer-pathan) [inline netbox/operations.py]: if possible provide limit parameter . No need to fetch more data for health check

File: netbox/playbooks/playbooks.json [hunk 8]
```diff
@@ -0,0 +1,2138 @@
...
+  "data": [
+    {
+      "uuid": "c88718e6-eae6-452e-be3c-6d3d1ca5fa09",
+      "@type": "WorkflowCollection",
+      "name": "Sample - NetBox - 1.0.0",
+      "description": "NetBox is the leading solution for modeling and documenting modern networks. By combining the traditional disciplines of IP address management (IPAM) and datacenter infrastructure management (DCIM) with powerful APIs and extensions, NetBox provides the ideal \"source of truth\" to power network automation.",
+      "visible": true,
+      "image": null,
+      "recordTags": [
+        "Netbox",
```
Comment (cs-sameer-pathan) [inline netbox/playbooks/playbooks.json]: change this tag to NetBox for all occurrences.

File: netbox/playbooks/playbooks.json [hunk 9]
```diff
@@ -0,0 +1,2138 @@
...
+            },
+            {
+              "uuid": "37a3d05b-fd2a-47cf-816f-eb766d0dc2d3",
+              "@type": "WorkflowStep",
+              "name": "Get IP Address List",
+              "description": null,
+              "status": null,
+              "arguments": {
+                "name": "NetBox",
+                "config": "''",
```
Comment (cs-sameer-pathan) [inline netbox/playbooks/playbooks.json]:
```suggestion
                "config": "",
```

**Review by cs-sameer-pathan — CHANGES_REQUESTED** [review]

**Comment by cs-adarsh-gupta** [comment]
Resolved comments

**Review by cs-prerna-joshi — APPROVED** [review]
LGTM

**Pull request merged**